systemtest/src/test/java/io/strimzi/systemtest/LogSettingST.java

```diff
@@ -479,6 +482,138 @@ void testJSONFormatLogging() {
+                        "rootLogger.appenderRef.console.ref = STDOUT\n" +
+                        "rootLogger.additivity = false"))
+                .build();
+
+        ConfigMap configMapUo = new ConfigMapBuilder()
+                .withNewMetadata()
+                .withName("external-configmap-uo")
+                .withNamespace(NAMESPACE)
+                .endMetadata()
+                .addToData(Collections.singletonMap("log4j2.properties", "name = UOConfig\n" +
+                        "\n" +
+                        "appender.console.type = Console\n" +
+                        "appender.console.name = STDOUT\n" +
+                        "appender.console.layout.type = PatternLayout\n" +
+                        "appender.console.layout.pattern = [%d] %-5p <%-12.12c{1}:%L> [%-12.12t] %m%n\n" +
+                        "\n" +
+                        "rootLogger.level = TRACE\n" +
+                        "rootLogger.appenderRefs = stdout\n" +
+                        "rootLogger.appenderRef.console.ref = STDOUT\n" +
+                        "rootLogger.additivity = false"))
+                .build();
+
+        kubeClient().getClient().configMaps().inNamespace(NAMESPACE).createOrReplace(configMapTo);
+        kubeClient().getClient().configMaps().inNamespace(NAMESPACE).createOrReplace(configMapUo);
+
+        ExternalLogging elTo = new ExternalLoggingBuilder()
+                .withName("external-configmap-to")
+                .build();
+
+        ExternalLogging elUo = new ExternalLoggingBuilder()
+                .withName("external-configmap-uo")
+                .build();
+
+        // change to external logging
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            k.getSpec().getEntityOperator().getTopicOperator().setLogging(elTo);
+            k.getSpec().getEntityOperator().getUserOperator().setLogging(elUo);
+        });
+
+        String finalEoPodName1 = eoPodName;
+        TestUtils.waitFor("Logger change", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT, () -> {
+                return cmdKubeClient().execInPodContainer(finalEoPodName1, "topic-operator", "cat", "/opt/topic-operator/custom-config/log4j2.properties").out().contains("rootLogger.level = TRACE")
+                        && cmdKubeClient().execInPodContainer(finalEoPodName1, "user-operator", "cat", "/opt/user-operator/custom-config/log4j2.properties").out().contains("rootLogger.level = TRACE");
+            }
+        );
+
+        eoPodName = eoPods.keySet().iterator().next();
+        eoPodHash = eoPods.get(eoPodName);
+        eoPods = DeploymentUtils.depSnapshot(eoName);
+        eoPodNameAfterLoggingChange = eoPods.keySet().iterator().next();
+        assertThat("Pod name changed after logging changed", eoPodNameAfterLoggingChange.equals(eoPodName), is(true));
+        assertThat("Pod hash changed after logging changed", eoPods.get(eoPodNameAfterLoggingChange).equals(eoPodHash), is(true));
```

Comment:
What kind of Hash is this comparing?

Reply:
Whatever `depSnapshot()` is returning. @Frawless could you enlight us?

Reply:
I assume you want to wait for rolling update of EO. You should use something like 
```DeploymentUtils.waitTillDepHasRolled(KafkaResources.entityOperatorDeploymentName(CLUSTER_NAME), 1, eoPods);```

Reply:
Oh no, I misunderstand few parts. If you want to check, that EO doesn't roll, you should store `DeploymentUtils.depSnapshot(eoName);` before you starting with any changes and compare the value with `DeploymentUtils.depSnapshot(eoName);` when the changes are applied. This method returns map with all pods connected to specific deployment and it's UIDs